src/targets/gpu/mlir.cpp

```diff
@@ -1096,6 +1096,37 @@ std::string dump_mlir(module m, const std::vector<shape>& inputs)
     return mlir_print(&mlirOperationPrint, mod_op);
 }
 
+std::string dump_mlir(module m, const std::vector<shape>& inputs, const fs::path& location)
+{
+    auto name =
+        mlir_program::get_symbol_name(m) + ".mlir";
+    replace_string_inplace(name, ", ", "_");
+    replace_string_inplace(name, ":", "s");
```

Comment:
This is missing the sizes. We should probably put this in a seperate function to compute the file name and reuse it for this function and `dump_mlir_to_mxr`, something like:

```cpp
static std::string compute_dump_name(const module& m, const std::string ext)
{
    std::vector<instruction_ref> sizes;
    for(auto ins : iterator_for(m))
    {
        if(not contains({"convolution", "dot"}, ins->name()))
            continue;
        sizes.insert(sizes.end(), ins->inputs().begin(), ins->inputs().end());
    }
    auto name =
        mlir_program::get_symbol_name(m) + "_" + shape::to_sizes_string(to_shapes(sizes)) + ext;
    replace_string_inplace(name, ", ", "_");
    replace_string_inplace(name, ":", "s");
    return name;
}
```

This will need to be called after `rewrite_reduce`.